etcd/etcd.go

```diff
@@ -86,13 +123,31 @@ func (r *EtcdAdapter) Register(service *bridge.Service) error {
 }
 
 func (r *EtcdAdapter) Deregister(service *bridge.Service) error {
-	path := r.path + "/" + service.Name + "/" + service.ID
-
 	var err error
-	if r.client != nil {
-		_, err = r.client.Delete(path, false)
+	if len(r.templates) < 1 {
+		// Default behavior if no templates are registered
+		path := r.path + "/" + service.Name + "/" + service.ID
+
+		if r.client != nil {
+			_, err = r.client.Delete(path, false)
+		} else {
+			_, err = r.client2.Delete(path, false)
+		}
 	} else {
-		_, err = r.client2.Delete(path, false)
+		toSet, err := r.executeTemplates(service)
+		if err == nil {
+			for key, value := range toSet {
```

Comment:
Fix for compile error "'value declared and not used'"

    -for key, value := range toSet {
    +for key := range toSet {